android/jni/OpenSLContext.cpp

```diff
@@ -82,27 +79,30 @@ bool OpenSLContext::Init() {
 	SLresult result;
 	// create engine
 	result = slCreateEngine(&engineObject, 0, NULL, 0, NULL, NULL);
-	if (result != SL_RESULT_SUCCESS) {
-		ERROR_LOG(AUDIO, "OpenSL: Failed to create the engine: %d", (int)result);
+	if (!CheckResult(result, "slCreateEngine")) {
 		engineObject = nullptr;
 		return false;
```

Comment:
If `Init()` returns false, the context (and error message) is deleted.

-[Unknown]

Reply:
Doh! I should have realized that I needed to store the error outside the context somewhere. Will put it in a global for now, and then one day I want to refactor this whole thing in some other way.